pkg/proxy/dialer.go

```diff
@@ -66,6 +81,11 @@ func ciliumDialer(identity int, network, address string) (net.Conn, error) {
 		return nil, fmt.Errorf("unable to connect: %s", err)
 	}
 
+	if err := setKeepAlive(c); err != nil {
```

Comment:
We also need to set those options on request connections, in `pkg/proxy/socket.go`'s `proxySocket.Accept`.

Reply:
fixed